src/rendergraph/opengl/backend/basenode.h

```diff
@@ -0,0 +1,70 @@
+#pragma once
+
+#include <QMatrix4x4>
+
+namespace rendergraph {
+class BaseNode;
+class Engine;
+} // namespace rendergraph
+
+class rendergraph::BaseNode {
+  public:
+    BaseNode() = default;
+    virtual ~BaseNode();
+
+    void setUsePreprocess(bool value) {
+        m_usePreprocess = value;
+    }
+    bool usePreprocess() const {
+        return m_usePreprocess;
+    }
+    virtual bool isSubtreeBlocked() const {
+        return false;
+    }
+    virtual void preprocess() {
+    }
+    virtual void render() {
+    }
+    virtual void initialize() {
+    }
+    virtual void resize(int, int) {
+    }
+    void setEngine(Engine* engine) {
+        m_pEngine = engine;
+    }
+    Engine* engine() const {
+        return m_pEngine;
+    }
+
+    // Prefer using NodeInterface<T>::appendChildNode(std::unique_ptr<BaseNode> pNode);
+    void appendChildNode(BaseNode* pChild);
+    // Prefer using std::unique_ptr<BaseNode> NodeInterface<T>::detachChildNode(BaseNode* pNode);
```

Comment:
Not sure to understand the doc-strings here - is there a reason for keeping `appendChildNode` and `removeChildNode` public?